InnerEye/ML/lightning_base.py

```diff
@@ -136,6 +139,27 @@ def setup(self) -> None:
         This hook reads the dataset file, and possibly sets required pre-processing objects, like one-hot encoder
         for categorical features, that need to be available before creating the model.
         """
+        # Following code validates segmentation training, validation and test data to ensure:
+        # 1) Files exist,
+        # 2) mask_id identifier is not empty,
+        # 3) consistency for input channels, ground_truth and mask_id,
+        # 4) ensures data is consistent with load_dataset_sources method prior running training, validation and testing.
+        if self.config.is_segmentation_model:
+            # Creates a list with all the channels of interest
+            all_channels = self.config.image_channels + self.config.ground_truth_ids
+            # Mask_id is an optional field. If non-empty in the config, will check dataframe.
+            if len(self.config.mask_id) > 0:
+                all_channels += [self.config.mask_id]
+            # Root directory where data is stored
+            local_dataset_root_folder = self.config.local_dataset
+            # Iterate over train, validation and test dataset
+            for split_data in [ self.config.get_dataset_splits().train, self.config.get_dataset_splits().val, self.config.get_dataset_splits().test]:
+                unique_ids = set(split_data[CSV_SUBJECT_HEADER])
+                for patient_id in unique_ids:
+                    rows = split_data.loc[split_data[CSV_SUBJECT_HEADER] == patient_id]
+                    # Validates condition so that data is consistent with
+                    __ = validated_channel_ids(all_channels, rows, local_dataset_root_folder, patient_id)
```

Comment:
Can you maybe wrap that up into one function that prints out ALL errors in one go? We don't want to put people in a position where the first run fails because of a problem with patient 2, they fix that, start again, then it fails again saying that there is a problem with patient 3.

Reply:
Implemented change: converts_channels_to_file_paths returns paths and string for missing channels and files for data frame. Error is raised and directed to standard error output if string converts_channels_to_file_paths is not empty -- snipped code below.

            for split_data in [ self.config.get_dataset_splits().train, self.config.get_dataset_splits().val, self.config.get_dataset_splits().test]:
                unique_ids = set(split_data[CSV_SUBJECT_HEADER])
                for patient_id in unique_ids:
                    rows = split_data.loc[split_data[CSV_SUBJECT_HEADER] == patient_id]
                    # Converts channels from data frame to file paths
                    __, failed_channel_info = converts_channels_to_file_paths(all_channels,
                                                                              rows,
                                                                              local_dataset_root_folder,
                                                                              patient_id)
                    full_failed_channel_info += failed_channel_info

        if full_failed_channel_info:
            raise ValueError(full_failed_channel_info)